src/blocks/overlay/edit.js

```diff
@@ -0,0 +1,282 @@
+import { __ } from '@wordpress/i18n';
+import { createBlock } from '@wordpress/blocks';
+import {
+	InnerBlocks,
+	InspectorControls,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import {
+	Button,
+	BaseControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+// eslint-disable-next-line import/no-unresolved
+import './editor.scss';
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @param {Object}   props               Block Props.
+ * @param {Object}   props.attributes    Block Attributes.
+ * @param {number}   props.clientId      Block ClientId.
+ * @param {Function} props.setAttributes Set Block Attributes.
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ *
+ * @return {Element} Element to render.
+ */
+export default function Edit( { attributes, clientId, setAttributes } ) {
+	const { PostSelectButton } = window.hm.components;
+
+	const { postId, postType } = attributes;
+
+	const [ contentExpanded, setContentExpanded ] = useState( false );
+
+	const childBlocks = useSelect(
+		( select ) =>
+			select( 'core/editor' ).getBlocksByClientId( clientId )[ 0 ]
+				.innerBlocks
+	);
+
+	const { media, post } = useSelect( ( select ) => {
+		// eslint-disable-next-line no-shadow
+		const post = select( 'core' ).getEntityRecord(
+			'postType',
+			postType,
+			postId
+		);
+
+		// eslint-disable-next-line no-shadow
+		const media = select( 'core' ).getMedia( post?.featured_media || 0 );
```

Comment:
Just an optimization - we don't need to make this API request if post.featured_media doesn't exist.